### ISSUE TITLE
Rule proposal: New Rule to Prevent Exporting Types and Named Exports in the Same Declaration

### ISSUE THREAD

**Issue by kevinwolfcr** [typescript-eslint/typescript-eslint#7121]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+new+plugin+rule%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] My proposal specifically checks TypeScript syntax, or it proposes a check that requires type information to be accurate.
- [X] My proposal is not a "formatting rule"; meaning it does not just enforce how code is formatted (whitespace, brace placement, etc).
- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Description

I'm working on a TypeScript project and have noticed that I would like to have a way to prevent exporting types and named exports in the same declaration. Specifically, I want to enforce separating type exports from other named exports.	

### Fail Cases

```typescript
export { foo, type Foo}
```

### Pass Cases

```typescript
export type { Foo }
export { foo }
```

**Comment by bradzacher** [comment]
https://typescript-eslint.io/rules/consistent-type-exports/

This rule will do some of that by default with its autofixer, but it won't enforce it is always top-level though.

We don'took to create bespoke rules to ban syntax unless there's significant community backing or substantial benefit we can provide with a standalone rule. I don't believe either apply here.
https://typescript-eslint.io/linting/troubleshooting#how-can-i-ban-specific-language-feature

You can just ban inline type exports.
